confd/pkg/backends/calico/client.go

```diff
@@ -899,8 +896,22 @@ func (c *client) onUpdates(updates []api.Update, needUpdatePeersV1 bool) {
 	// If configuration relevant to BGP peerings has changed, recalculate the set of v1
 	// peerings that should exist, and update the cache accordingly.
 	if needUpdatePeersV1 {
+		// Mark currently watched secrets as stale, so that they can be cleaned up if no
+		// longer needed.
+		if c.secretWatcher != nil {
+			c.secretWatcher.MarkStale()
+		}
+
 		log.Info("Recompute BGP peerings: " + strings.Join(needUpdatePeersReasons, "; "))
 		c.updatePeersV1()
+
+		// Also update BGP config passwords before removing old watched secrets
+		c.getNodeMeshPasswordKVPair(c.globalBGPConfig, model.GlobalBGPConfigKey{})
```

Comment:
We probably need to handle the case where `c.globalBGPConfig` is nil - I don't think we can assume that a BGP config exists on all clusters.

Reply:
Sure, I thought that there should always be a default BGP config, but it makes sense to do that check either way just in case there's a bug in the code with caching the config.